lib/commands/preload.ts

```diff
@@ -305,6 +304,31 @@ Can be repeated to add multiple certificates.\
 		}
 	}
 
+	private async checkImageFile(image: string, appDeviceType?: string) {
+		try {
+			const fs = await import('fs');
+			await fs.promises.access(image);
+		} catch (error) {
+			throw new ExpectedError(`Cannot access image file: "${image}"`);
+		}
+		const path = await import('path');
+		if (path.extname(image) === '.zip') {
+			const zipMsg =
+				'The image file must be unzipped unless the device type is Intel Edison';
+			if (appDeviceType && appDeviceType !== 'intel-edison') {
+				throw new ExpectedError(`Error: ${zipMsg}`);
```

Comment:
So basically the improvement is that, if we opportunistically know for a fact that the image file must not be a zip file, we abort rather than just warn. Further UX improvement might be to unzip the file for the user, but I'd say that this is already way better than infinite spinners or error messages about partition tables (when we can avoid them).

Reply:
Ah, the logic is still broken, isn't it? If I create an app of type QEMU X86 32bit with the web dashboard, then `appDeviceType` will be `qemux86`, but I am allowed to add devices of type `intel-edison` to that app, whose images would be Intel Edison zip files. So, basically, there are no circumstances in which the image file is necessarily a zip file or necessarily not a zip file, based solely on the application's default device type. In other words, the logic cannot rely on `appDeviceType` and there is no point obtaining it.

If you have saved a copy of your commit as it was yesterday, perhaps we should revert to it. Sorry for the mess. :-/

Reply:
`git reflog` knows all!